### DISCUSSION TITLE
Auto Update Total Price

### DISCUSSION THREAD

**Discussion by ismaeltj9** [filamentphp/filament#12597] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.70

### How can we help you?

Hello All,

My code is working as needed, but I am using ->hidden only to use it's callback and it react with live() of others fields, is there any other more apropriated function that is not hidden()? Thanks in advanced.

```php
Forms\Components\TextInput::make('price')
    ->live()
    ->required(),
Forms\Components\TextInput::make('quantity')
    ->live()
    ->required(),
Forms\Components\TextInput::make('total price')
    ->hidden(function (Get $get, Set $set) {
        $set('total price', (float)$get('price') * (float)$get('quantity'));
        return false;
    })
    ->disabled(),
```

**Comment by xalabama** [comment]
What about the afterStateUpdated method?

**Comment by ismaeltj9** [comment]
Hello @xalabama, afterStateUpdated would work if I put it in price or/and quantity, and after one of this update set 'total price', but not in the same place of example hidden, that wait price or quantity to set itself, I dont know why, but I tested.

**Reply by leandrocfe** [reply, marked as answer]
?
```php
Forms\Components\TextInput::make('price')
    ->live(onBlur: true)
    ->required(),
Forms\Components\TextInput::make('quantity')
    ->live(onBlur: true)
    ->required(),
Forms\Components\Placeholder::make('total price')
    ->content(fn(Forms\Get $get) => $get('price') * $get('quantity'))
```

**Reply by ismaeltj9** [reply]
Hello @leandrocfe , I didnt know about placeholder, it also react with live fields as hidden, the only problem is that it cannot save to database, I will think about my project and maybe I use this, thank so much for your help.
